transport/session/src/types.rs

```diff
@@ -204,9 +194,30 @@ impl Hash for SessionId {
     }
 }
 
+fn caps_to_ack_mode(caps: Capabilities) -> AcknowledgementMode {
+    if caps.contains(Capability::RetransmissionAck | Capability::RetransmissionNack) {
+        AcknowledgementMode::Both
+    } else if caps.contains(Capability::RetransmissionAck) {
+        AcknowledgementMode::Full
+    } else {
+        AcknowledgementMode::Partial
+    }
+}
```

Comment:
Should this not be `From<Capabilities> for AcknowledgementMode`?

Reply:
This is internal only just to this file, so I considered it redundant to implement the `From` trait.